src/compas/geometry/xforms/quaternion.py

```diff
@@ -0,0 +1,225 @@
+
+
+from compas.geometry.transformations import quaternion_multiply
+from compas.geometry.transformations import quaternion_conjugate
+from compas.geometry.transformations import quaternion_unitize
+from compas.geometry.transformations import quaternion_canonize
+from compas.geometry.transformations import quaternion_norm
+from compas.geometry.transformations import quaternion_is_unit
+
+from compas.geometry.basic import allclose
+
+__all__ = ['Quaternion']
+
+
+class Quaternion(object):
+
+    """Creates a ``Quaternion`` object.
```

Comment:
docstrings with lots of math and especially latex symbols can be marked as "raw" (r""") to avoid having to escape backslashes.